server/src/Infrastructure/Persistence/Interceptors/AuditableEntityInterceptor.cs

```diff
@@ -5,14 +5,20 @@ namespace StudyZen.Infrastructure.Persistence;
 
 public static class AuditableEntityInterceptor
 {
-    public static void SetCreateStamp(IAuditable instance)
+    public static void SetCreateStamp(object instance)
```

Comment:
Why do we need to take an object and then check if it's IAuditable? If all we're working with is an IAuditable instance, why let the function take anything else?

Reply:
If it took an IAuditable instead of object, we could not add it to the delegate in UnitOfWork.